WanderLost/WanderLost/Client/Components/Card.razor

```diff
@@ -1,26 +1,80 @@
-﻿<div class="card card_r--@(Merchant.Card.Rarity) @(ClassName)">
+﻿@inject MerchantHubClient HubClient
+
+<div class="card card_r--@(TopRarity) @(ClassName)">
     <div class="card__title">@(Merchant.Name)&nbsp;(@(Merchant.Zone))</div>
     <div class="card__location" style="background-image: url(/images/zones/@(Uri.EscapeDataString(Merchant.Zone)).jpg)" />
     <div class="stock">
         <div class="stock__title">Inventory:</div>
-        <div class="stock__item stock__item_r--@(Merchant.Card.Rarity)">
+        <div class="stock__item stock__item_r--@(CardRarity)">
             @(Merchant.Card.Name)
         </div>
-        <div class="stock__item stock__item_r--@(Merchant.Rapport.Rarity)">
+        <div class="stock__item stock__item_r--@(RapportRarity)">
             @(Merchant.Rapport.Name)
         </div>
     </div>
     <div class="card__buttons">
-        <CardButton Type="CardButton.Variant.Like" Active=true />
-        <div class="card__score">0</div>
-        <CardButton Type="CardButton.Variant.Dislike" Active=false />
+        <CardButton Type="CardButton.Variant.Like" OnClick="@(async () => await SubmitVote(VoteType.Upvote))" Active=true />
+        <div class="card__score">@Merchant.Votes</div>
+    <CardButton Type="CardButton.Variant.Dislike" OnClick="@(async () => await SubmitVote(VoteType.Downvote))" Active=false />
     </div>
 </div>
 
 @code {
-    [Parameter, EditorRequired]
-    public ActiveMerchant Merchant { get; set; }
 
-    [Parameter]
-    public string ClassName { get; set; }
+  public static Dictionary<Rarity, string> RarityMapping = new Dictionary<Rarity, string>()
+  {
+    {WanderLost.Shared.Data.Rarity.Normal, "normal"},
+    {WanderLost.Shared.Data.Rarity.Uncommon, "uncommon"},
+    {WanderLost.Shared.Data.Rarity.Rare, "rare"},
+    {WanderLost.Shared.Data.Rarity.Epic, "epic"},
+    {WanderLost.Shared.Data.Rarity.Legendary, "legendary"},
+    {WanderLost.Shared.Data.Rarity.Relic, "relic"}
+  };
+
+  [Parameter, EditorRequired]
+  public ActiveMerchant Merchant { get; set; }
+
+  [Parameter, EditorRequired]
+  public Dictionary<Guid, VoteType> Votes { get; set; }
+
+  [Parameter, EditorRequired]
+  public string Server { get; set; } = string.Empty;
```

Comment:
wouldn't it be easier to pass `onVoteUp` / `onVoteDown` callbacks as parameters instead of `Votes` + `Server` combination? I consider `Card` a low level component, so it'd be best to keep it as dumb as possible

Reply:
Did it a bit differently - gave `Card` callbacks for submitting votes + getting the current vote. Mostly since both require current Merchant ID, which is `Card`-specific, so it felt easier + more natural to have it like that